### ISSUE TITLE
Dataset

### ISSUE THREAD

**Issue by YanhaoZhang** [vagr8/R_Unet#1]
Hi, thanks a lot for releasing the code. May I ask if we need to preprocess the video downloaded from [kth dataset](https://www.csc.kth.se/cvap/actions/)? I download the walk video from this website, but it seems that the [following code](https://github.com/vagr8/R_Unet/blob/673335def4335494ae8f928f3d79180191457cea/train.py#L36-L43) in train.py does not perform correctly. Thanks a lot. 
``` python
cwd = os.getcwd()
os.chdir(cwd+video_path[1:])
dir_list = next(os.walk('.'))[1]
video_dir_list = []
for i in dir_list:
    i = video_path + str(i) + '/'
    video_dir_list.append(i)
os.chdir(cwd)

**Comment by Michael-MuChienHsu** [comment]
Hi.

1. Yes we have to preprocess the KTH video by slicing each video into frames.
2. About the code, I think the problem will be how you arrange your training data.
What I did is: I have a parent dir consists of multiple child dir, and each child dir consist of frames form the same video.

I hope this will help.

**Comment by YanhaoZhang** [comment]
@vagr8 Thanks a lot for your prompt reply! This is really helpful for me:). May I also ask if we need a ground truth for each image in the training phase, i.e., a binary image as UNet for training, or the input are just images from the video?

**Comment by Michael-MuChienHsu** [comment]
Yes, we need ground truth while training. Since, for example, 5 image inputs with 5 image predictions, the code will load 10 images at the start of each loop, and calculate loss for back propagation at the end of each loop.   

Please check the latest code I uploaded today, it has more detailed comments and it's faster.

**Comment by YanhaoZhang** [comment]
Thanks a lot for your detailed reply and the uploading of your latest code! Now I understand it. I will check your latest code. :)
